community/lucene-index/src/main/java/org/neo4j/index/impl/lucene/IndexType.java

```diff
@@ -67,7 +67,7 @@ public void addToDocument( Document document, String key, Object value )
         void removeFieldsFromDocument( Document document, String key, Object value )
         {
             Set<String> values = null;
-            if ( value != null )
+            if ( value != null && value.toString() != null )
```

Comment:
It would be great to avoid calling `toString()` using something like:

```
String stringValue;
if ( value != null && (stringValue = value.toString()) != null )
{
   ... use the stringValue here instead of calling toString() again
}
```

And likewise for all the other placed. WDYT?